### PR TITLE
ports: Add named pins to Arduino boards.

### PR DESCRIPTION
**Pull request by sebromero**
This PR adds missing pin names according to the pinout document on docs.arduino.cc and the corresponding identifiers in the Arduino cores. The goal is to have consistent pin names between Arduino and MicroPython.

### CODE REVIEW

**Comment by aliphys** [comment]
To be cross checked with `Pins_arduino.h` file of:
- [GIGA](https://github.com/arduino/ArduinoCore-mbed/blob/main/variants/GIGA/pins_arduino.h)
- [Nano 33 BLE](https://github.com/arduino/ArduinoCore-mbed/blob/main/variants/ARDUINO_NANO33BLE/pins_arduino.h)
- [Portenta C33](https://github.com/arduino/ArduinoCore-renesas/blob/main/variants/PORTENTA_C33/pins_arduino.h)
- [Nano RP2040 Connect](https://github.com/arduino/ArduinoCore-mbed/blob/main/variants/NANO_RP2040_CONNECT/pins_arduino.h)
- [Nano ESP32](https://github.com/espressif/arduino-esp32/blob/master/variants/arduino_nano_nora/pins_arduino.h)
- [Nicla Vision](https://github.com/arduino/ArduinoCore-mbed/blob/main/variants/NICLA_VISION/pins_arduino.h)
- [Portenta H7](https://github.com/arduino/ArduinoCore-mbed/blob/main/variants/PORTENTA_H7_M7/pins_arduino.h)

File: ports/esp32/boards/ARDUINO_NANO_ESP32/pins.csv [hunk 1]
```diff
@@ -12,6 +12,14 @@ D10,GPIO21
 D11,GPIO38
 D12,GPIO47
 D13,GPIO48
+D17,GPIO1
+D18,GPIO2
+D19,GPIO3
+D20,GPIO4
```
Comment (aliphys) [inline ports/esp32/boards/ARDUINO_NANO_ESP32/pins.csv:18]: ✅  Single digit GPIO pins are listed without a proceeding zero (e.g. `GPIO4` as opposed to `GPIO04`). This matches both:
- The [ESP 32](https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf) datasheet (pg 12):
image

- Our Arduino [Nano ESP 32 documentation](https://docs.arduino.cc/tutorials/nano-esp32/pin-setup/#nano-esp32-pin-map):
image

Comment (aliphys) [inline ports/esp32/boards/ARDUINO_NANO_ESP32/pins.csv:18]: ✅  Single digit GPIO pins are listed without a proceeding zero (e.g. `GPIO4` as opposed to `GPIO04`). This matches both:
- The [ESP 32](https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf) datasheet (pg 12):
image

- Our Arduino [Nano ESP 32 documentation](https://docs.arduino.cc/tutorials/nano-esp32/pin-setup/#nano-esp32-pin-map):
image

File: ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv [hunk 2]
```diff
@@ -1,3 +1,30 @@
+D2,P43
```
Comment (aliphys) [inline ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv]: Unclear to me what `P43` is in this context?

Looking at the `variant.cpp` file for the [Nano 33 BLE variant inside the mbed core](https://github.com/arduino/ArduinoCore-mbed/blob/68fbece79947ce2a64bf49f7ba5f5f8111d89454/variants/ARDUINO_NANO33BLE/variant.cpp#L82-L83), pin `D2` is defined as `P1_11`. This also matches with the
- **Altium schematic**, in which pin `D2` is connected to the GPIO39/P1.11 pin on the NINA-B306 module
image
- **[NINA-B306 datasheet](https://content.u-blox.com/sites/default/files/NINA-B3_DataSheet_UBX-17052099.pdf)**, in which `GPIO39` is labelled as `P1.11`
image

A similar observation is made for the rest of the pins. Maybe we have documented this somewhere, and I have missed it?

File: ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+A3,P29
+A4,P31
+A5,P2
+A6,P28
+A7,P3
+LEDR,P24
+LEDG,P16
+LEDB,P6
+LED_BUILTIN,P13
+LED_PWR,P41
```
Comment (aliphys) [inline ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv]: Another example, where it is not clear how specified pin links to the Altium schematics of the NINA-B306 module
image

File: ports/stm32/boards/ARDUINO_GIGA/pins.csv [hunk 4]
```diff
@@ -1,3 +1,110 @@
...
+D79,PB1
+D80,PC3
+D81,PC2
+D82,PC0
+D83,PA0
+D84,PA4
+D85,PA5
+D86,PI12
+D87,PJ13
+D88,PE3
```
Comment (aliphys) [inline ports/stm32/boards/ARDUINO_GIGA/pins.csv]: Added additional pins, based on the variant.cpp file for the [GIGA variant](https://github.com/arduino/ArduinoCore-mbed/blob/68fbece79947ce2a64bf49f7ba5f5f8111d89454/variants/GIGA/variant.cpp#L133-L140). These are present on the SPI header (J7).

```suggestion
D88,PE3
D89,PG9
D90,PD7
D91,PB3
D92,PA15
```

File: ports/stm32/boards/ARDUINO_GIGA/pins.csv [hunk 5]
```diff
@@ -1,3 +1,110 @@
...
+D93,PB5
+D94,PB13
```
Comment (aliphys) [inline ports/stm32/boards/ARDUINO_GIGA/pins.csv]: Added additional pins, based on the variant.cpp file for the [GIGA variant](https://github.com/arduino/ArduinoCore-mbed/blob/68fbece79947ce2a64bf49f7ba5f5f8111d89454/variants/GIGA/variant.cpp#L145-L155).

```suggestion
D94,PB13
D95,PI8
D96,PB10
D97,PH7
D98,PG3
D99,PA10
D100,PC13
```

**Review by aliphys — COMMENTED** [review]
Note:

- We recommend using CIPO/COPI/CS instead of MISO/MOSI/SS on our Docs: https://docs.arduino.cc/learn/communication/spi/ . However, Micropython uses the MISO/MOSI/SCK nomenclature according to their [documentation](https://docs.micropython.org/en/latest/esp32/quickref.html#software-spi-bus).

File: ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv (hunk 2 above)
Comment (sebromero) [reply ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv]: > Unclear to me what P43 is in this context?

It's described in the Jira task:

>  On NRF P1.0 starts at 32.

Comment (aliphys) [reply ports/nrf/boards/ARDUINO_NANO_33_BLE_SENSE/pins.csv]: So is it correct to assume that P1.11 turns to P1.43 (11+32) that is defined as P43 inside MicroPython?

File: ports/renesas-ra/boards/ARDUINO_PORTENTA_C33/pins.csv
Comment (aliphys) [inline ports/renesas-ra/boards/ARDUINO_PORTENTA_C33/pins.csv:1]: LGTM!

File: ports/rp2/boards/ARDUINO_NANO_RP2040_CONNECT/pins.csv
Comment (aliphys) [inline ports/rp2/boards/ARDUINO_NANO_RP2040_CONNECT/pins.csv:1]: We define pins `D0` and `D1` in `P1` and `P0` respectively in the mbed core. 
https://github.com/arduino/ArduinoCore-mbed/blob/68fbece79947ce2a64bf49f7ba5f5f8111d89454/variants/NANO_RP2040_CONNECT/variant.cpp#L13-L14

These correspond to GPIO0 and GPIO1 which are mapped to `TX` and `RX` respectively. Can we add:
```
D0,GPIO0
D1,GPIO1
```

File: ports/stm32/boards/ARDUINO_NICLA_VISION/pins.csv
Comment (aliphys) [inline ports/stm32/boards/ARDUINO_NICLA_VISION/pins.csv:1]: LGTM

File: ports/stm32/boards/ARDUINO_PORTENTA_H7/pins.csv
Comment (aliphys) [inline ports/stm32/boards/ARDUINO_PORTENTA_H7/pins.csv:1]: LGTM

**Pull request closed without merging**